coprocess_python.go

```diff
@@ -86,18 +84,20 @@ static void Python_HandleMiddlewareCache(char* bundle_path) {
 }
 
 static int Python_NewDispatcher(char* middleware_path, char* event_handler_path, char* bundle_paths) {
+	PyThreadState*  mainThreadState = PyEval_SaveThread();
+	gilState = PyGILState_Ensure();
```

Comment:
We were already using `PyEval_ReleaseLock` but this is the proper/better way of using it, and this replaces the lock operations.